icu4j/main/classes/core/src/com/ibm/icu/number/NumberFormatterSettings.java

```diff
@@ -552,6 +554,22 @@ public T usage(String usage) {
         return create(KEY_USAGE, usage);
     }
 
+    /**
+     * Specifies the {@code DisplayOptions}. Such as {@code GrammaticalCase} specifies the desired
+     * case for a unit formatter's output (e.g. accusative, dative, genitive).
+     *
+     * @return The fluent chain.
+     * @draft ICU 72
+     */
+    public T displayOptions(DisplayOptions displayOptions) {
+        // `displayCase` does not recognise the `undefined`
+        if (displayOptions.getGrammaticalCase() == GrammaticalCase.UNDEFINED) {
```

Comment:
Good idea, I am adding it right now.

Reply:
done

Reply:
Adding internal function is adding some burden to this PR because I need to create another file for the internal function in c. I added and the tests fails because I need to take care of many logistics. therefore, I removed it and I will added in a next PR.